certmanager/util.go

```diff
@@ -18,10 +18,40 @@ import (
 	"crypto"
 	"crypto/x509"
 	"encoding/pem"
+	"net/http"
 
 	"github.com/golang/glog"
 )
 
+const (
+	// The GCE metadata service URI to get identity token.
+	identityUri = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity"
+)
+
+// Get the GCE VM identity jwt token from its metadata server.
+// Note: this function only works in a GCE VM environment.
+func GetIdentityToken(aud string) ([]byte, error) {
```

Comment:
I don't feel `util.go` in `certmanager` package is the right place for this method. It will be confusing when reading `certmanager.GetIdentityToken()` as this method only returns the default ID token in GCP environment.

Maybe we can structure the code in a folder like `pkg/providers/gcp` and call the method as `gcp.GetDefaultIDToken()`. This allows us to have a similar support for non-GCP environment like aws